src/libraries/System.Runtime/tests/System.Reflection.Tests/AssemblyTests.cs

```diff
@@ -181,6 +181,19 @@ public void GetEntryAssembly()
             Assert.True(correct, $"Unexpected assembly name {assembly}");
         }
 
+        [Fact]
+        public void SetEntryAssembly()
+        {
+            Assembly? originalAssembly = Assembly.GetEntryAssembly();
+            Assert.NotNull(originalAssembly);
+
+            Assembly.SetEntryAssembly(null);
+            Assert.Null(Assembly.GetEntryAssembly());
+
+            Assembly.SetEntryAssembly(originalAssembly);
+            Assert.Equal(Assembly.GetEntryAssembly(), originalAssembly);
+        }
```

Comment:
Add a test case that verifies we throw for non-runtime assembly. You can create one of those e.g. by calling `new PersistedAssemblyBuilder(...)`.